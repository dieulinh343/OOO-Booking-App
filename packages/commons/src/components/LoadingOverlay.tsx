
import React from 'react';
import CenterLoading from './CenterLoading';

export function LoadingOverlay({
  text,
  spinnerSize,
}: {
  text: string;
  spinnerSize: string;
}) {
  return (
    <div className="u-positionAbsolute u-positionFull u-zIndex3">
      <div className="u-positionAbsolute u-positionFull u-backgroundWhite u-opacityThreeQuarter" />
      <CenterLoading
        text={text}
        hideText={!text}
        duration={1500}
        size={spinnerSize}
      />
    </div>
  );
}

export default LoadingOverlay;
