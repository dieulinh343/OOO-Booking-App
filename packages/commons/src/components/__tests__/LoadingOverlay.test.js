import React from 'react';
import { shallow } from 'enzyme';
import { LoadingOverlay } from '../LoadingOverlay';

describe('components/LoadingOverlay', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      text: 'sample text',
      spinnerSize: '100',
    };
    wrapper = shallow(<LoadingOverlay {...props} />);
  });

  it('should render successfully', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
