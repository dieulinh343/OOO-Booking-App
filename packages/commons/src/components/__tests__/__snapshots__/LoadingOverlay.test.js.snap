// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LoadingOverlay should render successfully 1`] = `
<div
  className="u-positionAbsolute u-positionFull u-zIndex3"
>
  <div
    className="u-positionAbsolute u-positionFull u-backgroundWhite u-opacityThreeQuarter"
  />
  <CenterLoading
    duration={1500}
    hideText={false}
    size="100"
    text="sample text"
  />
</div>
`;
